docs/topics/request-response.rst

```diff
@@ -219,6 +231,22 @@ different fields from different pages::
         item['other_url'] = response.url
         yield item
 
+::
+
+    def parse_page1(self, response):
+        item = MyItem()
+        item['main_url'] = response.url
+        request = scrapy.Request("http://www.example.com/some_page.html",
+                                 callback=self.parse_page2)
+        request.cb_kwargs['item'] = item
+        request.cb_kwargs['foo'] = 'bar'
+        yield request
+
+    def parse_page2(self, response, item, foo):
+        item['other_url'] = response.url
+        item['foo'] = foo
```

Comment:
I see a slight problem with this example: It would have made more sense to define `item['foo'] = foo` on `parse_page1`, since you are already instantiating an item.

I’m thinking we could switch to something like:

```
    def parse_page1(self, response):
        # You can pass a cb_kwargs dictionary to a Request constructor:
        request = scrapy.Request("http://www.example.com/some_page.html",
                                 callback=self.parse_page2,
                                 cb_kwargs={'main_url': response.url})
        # You can alternatively modify the cb_kwargs property of an existing Request object:
        request.cb_kwargs['title'] = response.css('title::text').get()
        yield request

    def parse_page2(self, response, title, main_url):
        yield {'title': title, 'main_url': main_url, 'other_url': response.url}
```